python/pyarrow/_dataset.pyx

```diff
@@ -436,6 +474,14 @@ cdef class Fragment:
         self.init(sp)
         return self
 
+    cdef inline shared_ptr[CFragment] unwrap(self):
+        return self.wrapped
+
+    @property
+    def schema(self):
+        """Return the schema of batches scanned from this Fragment."""
```

Comment:
I am bit hesitant about adding a `schema` property, giving our discussion about "physical" schema vs "reader" schema, and it is not directly clear which of the two this would be (the docstring explains it, but just from the name)

But let's deal with that in a possible follow-up

Reply:
Yes, this ambiguity will be addressed in ARROW-8065